examples/dreamcast/cd/sound-dma/sound-dma.cpp

```diff
@@ -0,0 +1,67 @@
+#include <stdio.h>
+
+#include <unistd.h>
+#include <sys/ioctl.h>
+
+#include <dc/cdrom.h>
+#include <arch/cache.h>
+
+#define AICA_REG(n) (*(volatile uint32_t*)(0xA0700000+n))
+#define GD_PROTECTION  (*(volatile uint32_t*)0xA05F74B8)
+#define SYSCALLS_32(n) (*(volatile uint32_t*)(0xAC000000 + n))
+
+int main(int argc, char *argv[]) {
+
```

Comment:
```suggestion
    /* If the face buttons are all pressed, exit the app */
    cont_btn_callback(0, CONT_RESET_BUTTONS, on_reset);
```
All of the examples are required to have a standard way to exit them from the controller, so we can easily return to the bootloader. This line will make it so when you push all the face buttons on the controller + start (reset sequence for retail games), the `on_reset` callback will be called... Which you could just make:
```
static void on_reset(uint8_t addr, uint32_t btns) {
    (void)addr; (void)btns;
    quit = true;
}
```
and then make your main loop basically be `while(!quit) {}`. Check this example for a good reference: https://github.com/KallistiOS/KallistiOS/blob/master/examples/dreamcast/vmu/vmu_beep/beep.c.